group-coordinator/src/main/java/org/apache/kafka/coordinator/group/assignor/PartitionAssignor.java

```diff
@@ -0,0 +1,43 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.coordinator.group.assignor;
+
+import org.apache.kafka.common.annotation.InterfaceStability;
+
+/**
+ * Server side partition assignor used by the GroupCoordinator.
```

Comment:
isn't this used by the new client side assignor as well? 
> The new PartitionAssignor interface will be introduced to replace the ConsumerPartitionAssignor interface

also, the kip mentions that we'll be adding uniform & range for server side assignor. what about client side assignors?

Reply:
Not this one. We will introduce another one for the client. They are close but different and we want to evolve them separately in the future.

Reply:
The difference here is the packages they will be in?

Reply:
That’s right. We could perhaps name this one GroupCoordinatorPartitionAssignor to make the difference clearer. Having the same name seems confusing. What do you think?

Reply:
Yeah I was just thinking about that, but didn't know if the name would be awkward. Thinking about it though, it might be better to clarify through naming to save headaches in the future.

Reply:
I think that I will keep the current name for now. I am not fully convinced by `GroupCoordinatorPartitionAssignor`....